kartothek/io/dask/bag_cube.py

```diff
@@ -39,35 +44,38 @@
 
 
 def build_cube_from_bag(
-    data,
-    cube,
-    store,
-    ktk_cube_dataset_ids=None,
-    metadata=None,
-    overwrite=False,
-    partition_on=None,
-):
+    data: db.Bag,
+    cube: Cube,
+    store: Callable[[], KeyValueStore],
```

Comment:
FYI We have a type alias `StoreFactory` for this in `kartothek.core.typing`

Reply:
Thanks for pointing this out. The type hints are overall quite confusing because sometimes store is a callable and sometimes not. I have simply followed the convention from the previous doc strings.

Reply:
Indeed, this is very confusing. With https://github.com/JDASoftwareGroup/kartothek/pull/345 I tried to change the behaviour such that user facing code just doesn't care but this was not done for the cubes and everything is overall pretty messy